providers/implementations/kdfs/hkdf.c

```diff
@@ -126,6 +127,46 @@ static void kdf_hkdf_reset(void *vctx)
     ctx->provctx = provctx;
 }
 
+static int dm(const void *src, size_t src_len,
+              unsigned char **dest, size_t *dest_len)
+{
```

Comment:
This isn't a good pattern to use - if src is null then we still get a positive return value and dest and dest_len are untouched  - which they really should be set to null and 0 respectively so on success return the output values are logically correct (rather than untouched).

Reply:
And given you are adding this same static function all over the place it should be a common routine.

Reply:
This is only ever called after a context new.  Setting them to NULL and 0 would be clearing the already cleared.

If this function were made a provider utility function (which is quite tempting since it's duplicated several times), I'd agree.  But as it stands, I don't see the point.

Reply:
Missed your second note.  I've made it into a common routine and added the extra zeroing.